### PR TITLE
[docs] fix README.raspberrypi to make depends build succeed on Ubuntu 16.04

### PR DESCRIPTION
**Pull request by asavah**
## Description
Added minimal changes to README.raspberrypi to make the depends build succeed
on Ubuntu 16.04.
14.04 is of no use because its ancient g++ lacks proper regex support.
Also added bison and make to apt-get install list because these packages are missing
on the default ubuntu 16.04.4 server install.

## Motivation and Context
Using the current README.raspberrypi as is won't lead to a successful build.

## How Has This Been Tested?
Successfully built kodi using depends system on Ubuntu 16.04 x64. 

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: docs/README.raspberrypi [hunk 1]
```diff
@@ -38,7 +38,7 @@ build for the first Raspberry Pi, the commands have to be adapted to use
        --with-toolchain=$RPI_DEV/tools/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf \
        --with-firmware=$RPI_DEV/firmware \
        --with-platform=raspberry-pi2 \
-       --build=i686-linux \
+       --build=x86_64-linux-gnu \
```
Comment (Rechi) [inline docs/README.raspberrypi]: This argument isn't needed anymore.
It was necessary as the rbpi toolchain had i686 only executables at the beginning.

Comment (asavah) [reply docs/README.raspberrypi]: Let me check if it builds without it.
If memory serves some packages were picky about not having both --host and --build set.

**Comment by asavah** [comment]
@Rechi builds fine without --build line too, updated and squashed.

**Comment by hudokkow** [comment]
@asavah, I would really be grateful if you could test https://github.com/hudokkow/xbmc/blob/The_X_Files/docs/README.RaspberryPi.md. It's part of https://github.com/xbmc/xbmc/pull/14072.

I'm missing `bison` on my guide. `make` is part of `build-essential`.

**Pull request merged**